src/xdist/workermanage.py

```diff
@@ -252,9 +252,9 @@ def setup(self):
             args = make_reltoroot(self.nodemanager.roots, args)
         if spec.popen:
             name = "popen-%s" % self.gateway.id
-            if hasattr(self.config, "_tmpdirhandler"):
-                basetemp = self.config._tmpdirhandler.getbasetemp()
-                option_dict["basetemp"] = str(basetemp.join(name))
+            basetemp = self.config.getoption("basetemp")
```

Comment:
this only fixes half of the problem, pytest will undo the removal of _tmpdirhandler until we come up with a good full solution

Reply:
Oh, did I miss the issue which contained those information? What is the other half?

Thanks for the quick response!

Reply:
Ah, found it: https://github.com/pytest-dev/pytest-xdist/issues/510. I'll close this unless I get more guidance for the other part of the solution.